script/run-gn

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python
+
+import argparse
+import json
+import os
+import sys
+
+import lib.gn as gn
+
+from lib.config import get_output_dir, COMPONENTS, \
+                       SOURCE_ROOT, SRC_DIR, DEPOT_TOOLS_DIR
+
+
+# TODO(alexeykuzmin): There are no reasons to keep args files
+# in the "chromiumcontent" folder.
+CHROMIUMCONTENT_SOURCE_DIR = os.path.join(SOURCE_ROOT, 'chromiumcontent')
+
+
+def parse_args():
+  parser = argparse.ArgumentParser(description='Generate Ninja')
+
+  parser.add_argument('-a', '--args', type=json.loads, default=None,
+                      help='List of GN args in JSON format. E.g. \'{"key": "value"}\'')
+  parser.add_argument('-c', '--component', nargs='+', default=None, choices=COMPONENTS,
+                      help='Component(s) to generate build files for.')
+  parser.add_argument('-t', '--target_arch', default='x64', help='E.g. "x64".')
+
+  return parser.parse_args()
+
+
+def main():
+  args = parse_args()
+
+  # Generate build files for all components if none are specified.
+  components = args.component if args.component else COMPONENTS
+  target_arch = args.target_arch
+
+  # List of common default GN config args. Please keep in alphabetical order.
+  # Any value can be overridden by the '--args' script argument.
+  gn_args = {
+    'target_cpu': get_target_cpu(target_arch),
+    'target_os': get_target_os()
+  }
+  # Add new and override default args from a command line argument.
+  if args.args:
+     gn_args.update(args.args)
+
+  for component in components:
+    generate_build_files(component, target_arch, **gn_args)
+
+
+def generate_build_files(component, target_arch, **gnargs):
+  build_dir = create_build_dir(component, target_arch)
```

Comment:
i don't think GN requires the build dir to exist before running `gn gen`, so you might be able to skip this bit

Reply:
`gn gen` doesn't require the build dir to exist but we write `args.gn` file to the build dir before running `gn gen` for it, so the dir should be there anyway.
Moved dir creation to the `lib/gn.py`.